src/storage/src/hummock/sstable/in_memory_iterator.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_hummock_sdk::VersionedComparator;
+
+use crate::hummock::iterator::{Forward, HummockIterator, ReadOptions};
+use crate::hummock::sstable_store::TableHolder;
+use crate::hummock::table_accessor::StorageTableAcessor;
+use crate::hummock::value::HummockValue;
+use crate::hummock::{BlockHolder, BlockIterator, HummockResult, SSTableIteratorType};
+use crate::monitor::StoreLocalStatistic;
+
+pub struct InMemoryTableIterator {
```

Comment:
I am thinking whether we can unify the prefetched sstable read and block-by-block sstable read. For example, given a `Sstable`, we first check its `blocks` field. If it's Some(blocks), we read from the block directly, and if it's None, we read as what the current code does. In this way, we don't need such a separate `InMemoryTableIterator`.

Reply:
`InMemoryTableIterator ` has been removed.